src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -910,3 +936,91 @@ def firewall_rule_update(
         resource_group_name=resource_group_name,
         start_ip_address=start_ip_address or instance.start_ip_address,
         end_ip_address=end_ip_address or instance.end_ip_address)
+
+#####
+#           sql server key
+#####
+
+def server_key_create(
+        client,
+        resource_group_name,
+        server_name,
+        key_id=None):
+    
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.create_or_update(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name,
+        parameters=ServerKey(
+            server_key_type=ServerKeyType.azure_key_vault.value,
+            uri=key_id
+        )
+    )
+
+def server_key_get(
+        client,
+        resource_group_name,
+        server_name,
+        key_id):
+
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.get(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name)
+
+def server_key_delete(
+        client,
+        resource_group_name,
+        server_name,
+        key_id):
+
+    key_name = get_server_key_name_from_uri(key_id)
+
+    return client.delete(
+        resource_group_name=resource_group_name,
+        server_name=server_name,
+        key_name=key_name)
+
+def get_server_key_name_from_uri(uri):
+
+    match = re.match('https(.)+\.vault(.)+\/keys\/[^\/]+\/[0-9a-zA-Z]+$', uri)
```

Comment:
should there be a `^` at the start of this regex to match the start of the string? so that you don't match a string with leading junk like `"blah blha hello world https://YourVaultName.vault.azure.net/keys/YourKeyName/01234567890123456789012345678901"`